mantidproject/mantid
# Indirect Diffraction OSIRIS Spectra Range

Fixes #13850 

OSIRIS Diffraction Reduction now has spectra range enabled.
# To Test (set up)
- Open Diffraction (Interfaces > Indirect > Diffraction)
- Change Instrument to:
  - Instrument = OSIRIS
  - Analyser = diffraction
  - Reflection = diffonly
- Add input & Calibration
  - _All files are available from `\ExternalData\Testing\Data\UnitTest\`_
  - Run Numbers = `OSI10203.raw`
  - Cal File = `osiris_041_RES10.cal`
  - Vanadium Runs = `OSI10156.raw`
# To Test (Good detector range)
- Follow steps in `set up` section above.
- Spectra Min = 3
- Spectra Max = 361
- Click `Run`
- Ensure that the algorithm runs correctly and produces 2 workspaces:
  - `OSI10203_dRange`
  - `OSI10203_tof`
# To Test (Bad detector range)
- Follow steps in `set up` section above
- Spectra Min = 16
- Spectra Max = 25
- Click `Run`
- Ensure that a run time error is thrown and the message is displayed in the Results log stating:

**Error in execution of algorithm OSIRISDiffractionReduction
No selected Detectors found in .cal file for input range. Please ensure spectra range has atleast one selected detector.**

The unit tests have also been updated for this issue so please ensure these are run on the build servers and pass
